### PR TITLE
fix: Ensure proper resource management by using .use for file operations [WPB-11908]

### PR DESCRIPTION
**Pull request by kl3jvi**
----
#### PR Submission Checklist for internal contributors

- refactor(code-generation): Use Kotlin's `use` function for file writing  
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    I have filled the relevant parts of the PR description template to the best of my ability. Here's a summary of the changes in this pull request.
----

# What's new in this PR?

### Issues

This PR addresses a minor inefficiency in the way files are written in the code generation process. Previously, the code manually opened an output stream, wrote to it, and then closed it. This approach can be prone to errors if an exception occurs before the stream is closed, potentially leading to resource leaks.

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

This PR refactors the code to use Kotlin's `use` function for file writing. The `use` function ensures that the output stream is automatically closed when the block completes, even if an exception is thrown. This makes the code more concise, safer, and less prone to resource leaks.

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by vitorhugods** [comment]
Hi! Thanks for contributing :)

We don't get PRs from people outside the @wireapp organisation often, and our CI has been constantly evolving. So it's expected that some checks might fail because we haven't tested PRs from forks.

Because of that, please ignore failures on `add-jira-description`, and `semantic-commit-pr-title-lint`.

I'll check what happened with the Smoke Tests on Monday :)

**Comment by kl3jvi** [comment]
Hey @vitorhugods I plan on contributing some more here so I'm gonna stick around

**Comment by kl3jvi** [comment]
@vitorhugods any update on this

File: ksp/src/main/kotlin/com/wire/android/di/ViewModelScopedPreview.kt [hunk 1]
```diff
@@ -76,9 +73,8 @@ internal class ViewModelScopedPreviewProcessor(private val codeGenerator: CodeGe
                 "import ${item.qualifiedName?.asString()}\n\n" +
                 "data object $name : ${item.simpleName.asString()}"
         val dependencies = Dependencies(aggregating = true, *listOfNotNull(item.containingFile).toTypedArray())
-        val file: OutputStream = codeGenerator.createNewFile(dependencies, packageName, name, "kt")
-        file.write(content.toByteArray())
-        file.close()
+        codeGenerator.createNewFile(dependencies, packageName, name, "kt")
+            .use { it.write(content.toByteArray()) }
```
Comment (MohamadJaara) [inline ksp/src/main/kotlin/com/wire/android/di/ViewModelScopedPreview.kt:75]: i wonder if the same is needed in the next function at line 89 ?

Comment (kl3jvi) [reply ksp/src/main/kotlin/com/wire/android/di/ViewModelScopedPreview.kt:75]: You are absolutely right

Comment (kl3jvi) [reply ksp/src/main/kotlin/com/wire/android/di/ViewModelScopedPreview.kt:75]: @MohamadJaara Can you have a look again, I implemented the changes :)

**Comment by MohamadJaara** [comment]
@kl3jvi can you please run `./gradlew detektAll` and fix any issues that come out of it?

**Comment by kl3jvi** [comment]
> @kl3jvi can you please run `./gradlew detektAll` and fix any issues that come out of it?

Just fixed

**Comment by kl3jvi** [comment]
@MohamadJaara Can you check please, the pipeline failed bcz an aws key

**Review by MohamadJaara — APPROVED** [review]

**Review by vitorhugods — APPROVED** [review]

**Pull request merged**